File: apps/api/app/models/esl_device.py
import uuid
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class ESLDevice(Base):
    __tablename__ = "esl_devices"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True)
    dealership_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("dealerships.id", ondelete="CASCADE"), nullable=False
    )
    device_id: Mapped[str] = mapped_column(Text, nullable=False)
    provider: Mapped[str | None] = mapped_column(Text)
    provider_device_id: Mapped[str | None] = mapped_column(Text)
    model: Mapped[str | None] = mapped_column(Text)
    screen_width: Mapped[int | None] = mapped_column(Integer)
    screen_height: Mapped[int | None] = mapped_column(Integer)
    battery_level: Mapped[int | None] = mapped_column(Integer)
    signal_status: Mapped[str | None] = mapped_column(Text)
    gateway_id: Mapped[str | None] = mapped_column(Text)
    status: Mapped[str | None] = mapped_column(Text)
    last_seen_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    last_updated_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

File: apps/web/src/lib/types/esl-device.ts
export interface ESLDevice {
  id: string;
  dealership_id: string;
  device_id: string;
  provider: string | null;
  provider_device_id: string | null;
  model: string | null;
  screen_width: number | null;
  screen_height: number | null;
  battery_level: number | null;
  signal_status: string | null;
  gateway_id: string | null;
  status: string | null;
  last_seen_at: string | null;
  last_updated_at: string | null;
  created_at: string;
}
